Formula/crytic-compile.rb

```diff
@@ -38,18 +39,14 @@ def install
   end
 
   test do
-    (testpath/"test.sol").write <<~EOS
-      pragma solidity ^0.8.0;
-      contract Test {
-        function f() public pure returns (bool) {
-          return false;
-        }
-      }
-    EOS
-
-    system "solc-select", "install", "0.8.0"
-    with_env(SOLC_VERSION: "0.8.0") do
-      system bin/"crytic-compile", testpath/"test.sol", "--export-format=solc", "--export-dir=#{testpath}/export"
+    resource "testdata" do
```

Comment:
@Homebrew/core is this a thing we're doing now? Not opposed, kinda in favour if it saves a non-test-time download, but just wanted to check it 1) works and 2) is desirable.

Reply:
By "this", do you mean `resource` blocks inside `test` blocks? IIRC there were a handful of examples of this a while back that we had to get rid of to enable formulae being frozen after instantiation.

Reply:
Seems to work, though, so I must be misremembering. Or maybe formulae freezing has changed since then. Probably the former. @Bo98?

Reply:
> By "this", do you mean resource blocks inside test blocks? 

@carlocab Yup!

Reply:
Just to add context, I opted to include this resource block inside the test block to avoid having `virtualenv_install_with_resources` trying to install the test file as a python package.